### ISSUE TITLE
Kosmos 2.5

### ISSUE THREAD

**Issue by conceptofmind** [microsoft/unilm#1297]
Hello,

Thank you for all your great research. 

I was wondering if there were plans to release Kosmos 2.5 similar to how Kosmos 2 was released on Huggingface.

Thank you,

Enrico

**Comment by pribadihcr** [comment]
+1

**Comment by logan-markewich** [comment]
+1

**Comment by FrancescoSaverioZuppichini** [comment]
+1

**Comment by michaelhadoop** [comment]
+1

**Comment by fernando-indrema** [comment]
+1

**Comment by evolu8** [comment]
please +1

**Comment by luvwinnie** [comment]
+1

**Comment by shenguojun** [comment]
+1

**Comment by MiaSanLei** [comment]
+1

**Comment by dydxdt** [comment]
+1

**Comment by coder4nlp** [comment]
+1

**Comment by zykzyk1993** [comment]
+1 please

**Comment by ydshieh** [comment]
+1 please 🙏

**Comment by acul3** [comment]
+ 1

**Comment by contrebande-labs** [comment]
Would be nice if there was an official acknowledgement of this.

**Comment by Natyren** [comment]
Hi @wolfshow, сould you please let people there know if there are plans to release the code and weights for this model?

**Comment by wolfshow** [comment]
Thank you all for expressing interest in Kosmos-2.5. We intend to release the code and models, although we do not have a specific ETA at this time. Our goal is to ensure that the models are exceptionally user-friendly, and as such, there is still considerable work to be completed before we make them open source. We appreciate your patience and look forward to sharing our progress with you.

**Comment by ydshieh** [comment]
Thank you @wolfshow  for the information. Super glad your team considers to open source this exciting model!

**Comment by FrancescoSaverioZuppichini** [comment]
> Thank you all for expressing interest in Kosmos-2.5. We plan to release the code and models, though we currently do not have a specific ETA. It's important to note that we are currently facing limitations in GPU resources. Despite this challenge, we are committed to making the models extremely user-friendly. There is still a significant amount of work ahead of us before we can open source the project, and we appreciate your understanding and patience as we navigate these constraints. We look forward to updating you on our progress.

Thanks a lot for breaking the silence :) I have to wonder, the paper did shows result that result must have come from some code and some trained model. Why do you need to use GPUs now? Would it be possible to release what you have done for the paper and let the community take care of the rest? Otherwise, it is very hard for us to validate the results and also to take advantage of the research done. I am sure all the authors would love to see people using their work.

Thank you again

**Comment by logan-markewich** [comment]
@FrancescoSaverioZuppichini it's microsoft research, so I'm sure they have their own red tape, responsibilities, and procedures to adhere too :)

**Comment by FrancescoSaverioZuppichini** [comment]
> @FrancescoSaverioZuppichini it's microsoft research, so I'm sure they have their own red tape, responsibilities, and procedures to adhere too :)

Sure, They could have said it weeks ago if they cared to share them :) Well looks like this discussion as come to an end.

**TL;DR** At some point hopefully when the model is still relevant, we'll get some code that for some reason needs GPUs time while the paper showed results that must have been run on GPUs. We are living in the matrix

**Comment by wolfshow** [comment]
> > @FrancescoSaverioZuppichini it's microsoft research, so I'm sure they have their own red tape, responsibilities, and procedures to adhere too :)
> 
> Sure, They could have said it weeks ago if they cared to share them :) Well looks like this discussion as come to an end.
> 
> **TL;DR** At some point hopefully when the model is still relevant, we'll get some code that for some reason needs GPUs time while the paper showed results that must have been run on GPUs. We are living in the matrix

We needs computations to run more experiements which is not included in the paper right now.

**Comment by evolu8** [comment]
Thank you @wolfshow . Your efforts are very much appreciated! Good luck with trying off the final pieces.

**Comment by shubhamagarwal92** [comment]
Hi @wolfshow! 
Interesting work on Kosmos 2.5! 

Is there any plan to release the datasets as well? Or the corresponding pipeline code for different datasets (Sec 2.3/2.4 in the paper)?

**Comment by evolu8** [comment]
@wolfshow  - any news?

**Comment by evolu8** [comment]
@wolfshow - another month, another nudge :) Please keep us posted.

**Comment by ydshieh** [comment]
Really hope this could be released - it would have huge impact 🙏 🔥

**Comment by coder4nlp** [comment]
Please!

**Comment by FrancescoSaverioZuppichini** [comment]
Hard F

**Comment by atlury** [comment]
Any ETA for the code and models for this paper? Its been nearly 3-4 months.

**Comment by radiachkik** [comment]
+1

**Comment by luohao123** [comment]
Still plan to release the code? The opensource community really needs such a work!

**Comment by sairin94** [comment]
+1

**Comment by AlexAndrei98** [comment]
+11 🚀 🚀 🚀

**Comment by wolfshow** [comment]
Kosmos 2.5 models and code have been publicly available at https://huggingface.co/microsoft/kosmos-2.5 and https://github.com/microsoft/unilm/tree/master/kosmos-2.5. Thanks!

**Comment by wolfshow** [comment]
> Hello,
> 
> Thank you for all your great research.
> 
> I was wondering if there were plans to release Kosmos 2.5 similar to how Kosmos 2 was released on Huggingface.
> 
> Thank you,
> 
> Enrico

https://github.com/huggingface/transformers/issues/30877

**Comment by EwoutH** [comment]
Maybe notable: Kosmos-2.5 now also is published in [Safetensors](https://huggingface.co/docs/safetensors/en/index) format [on HuggingFace](https://huggingface.co/microsoft/kosmos-2.5).

**Comment by atlury** [comment]
Ok an example with image input on hugging spaces would be highly helpful.
